datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -785,6 +828,35 @@ mod tests_statistical {
         }]
     }
 
+    /// Create join filter for NLJoinExec with expression `big_col > small_col`
+    /// where both columns are 0-indexed and come from left and right inputs respectively
+    fn nl_join_filter() -> Option<JoinFilter> {
+        let column_indices = vec![
+            ColumnIndex {
+                index: 0,
+                side: JoinSide::Left,
+            },
+            ColumnIndex {
+                index: 0,
+                side: JoinSide::Right,
+            },
+        ];
+        let intermediate_schema = Schema::new(vec![
+            Field::new("big_col", DataType::Int32, false),
+            Field::new("small_col", DataType::Int32, false),
+        ]);
+        let expression = Arc::new(BinaryExpr::new(
+            Arc::new(Column::new_with_schema("big_col", &intermediate_schema).unwrap()),
+            Operator::Gt,
+            Arc::new(Column::new_with_schema("big_col", &intermediate_schema).unwrap()),
```

Comment:
the comments say `big_col > small_col` but this test seems to be doing `big_col > big_col`

Reply:
Thank you! Fixed.

(Didn't affect tests, since reordering influenced only by left/right inputs and their statistics, so filter expression doesn't matter)